### DISCUSSION TITLE
Compatibility of bucket4j-spring-boot-starter

### DISCUSSION THREAD

**Discussion by ShriramDeshkar** [MarcGiffing/bucket4j-spring-boot-starter#227] [Q&A]
Hi ,

I m using bucket4j-spring-boot-starter with version 0.10.2, with redisson . Currently I m using redisson-spring-boot-starter with version 3.17.0. And spring-boot-starter-parent version is 3.2.2 . 
With this versions my application is working fine . But as soon as I upgrade redisson-spring-boot-starter version my spring boot application is not working.
So which is compatible version for redisson-spring-boot-starter for latest bucket4j-spring-boot-starter version of 0.10.2

**Comment by MarcGiffing** [comment]
Can you provide an example project? It would help to solve your issue.

**Reply by ShriramDeshkar** [reply]
Hi, Thanks for response 
Below is my use case 
I m using azure cache for redis instance 


import javax.cache.CacheManager;
import javax.cache.Caching;
import org.redisson.config.Config;
import org.redisson.jcache.configuration.RedissonConfiguration;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import io.github.bucket4j.distributed.proxy.ProxyManager;
import io.github.bucket4j.grid.jcache.JCacheProxyManager;
@Configuration
public class RedisConfig {

	@Bean
    public Config config() {
        Config config = new Config();
        config.useSingleServer().setPassword(System.getenv("REDIS_CACHE_KEY"))
                .setAddress("rediss://"+ System.getenv("REDIS_CACHE_HOST_NAME") +":6380");
        return config;
    }

    @Bean(name = "springCM")
    public CacheManager cacheManager(Config config) {
        CacheManager manager = Caching.getCachingProvider().getCacheManager();
        manager.createCache("cache", RedissonConfiguration.fromConfig(config));
        manager.createCache("userList", RedissonConfiguration.fromConfig(config));
        return manager;
    }

    @Bean
    ProxyManager<String> proxyManager(CacheManager cacheManager) {
        return new JCacheProxyManager<>(cacheManager.getCache("cache"));
    }
}


and initially my POM.xml contains below dependencies

		<dependency>
			<groupId>org.redisson</groupId>
			<artifactId>redisson-spring-boot-starter</artifactId>
			<version>3.17.0</version>
		</dependency>
		
		<dependency>
			<groupId>com.giffing.bucket4j.spring.boot.starter</groupId>
			<artifactId>bucket4j-spring-boot-starter</artifactId>
			<version>0.5.2</version>
			<exclusions>
				<exclusion>
					<groupId>org.ehcache</groupId>
					<artifactId>ehcache</artifactId>
				</exclusion>
			</exclusions>
		</dependency>

Now when I changed version of redisson-spring-boot-starter version to 3.26.0 or even 3.18.0 & bucket4j-spring-boot-starter version of 0.10.2
it is trying to connect to redis before my cache redis starts its instance.

The error I m getting is as below
Error creating bean with name 'redisTemplate' defined in class path resource [org/redisson/spring/starter/RedissonAutoConfiguration.class]: Unsatisfied dependency expressed through method 'redisTemplate' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'redissonConnectionFactory' defined in class path resource [org/redisson/spring/starter/RedissonAutoConfiguration.class]: Unsatisfied dependency expressed through method 'redissonConnectionFactory' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'redisson' defined in class path resource [org/redisson/spring/starter/RedissonAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.redisson.api.RedissonClient]: Factory method 'redisson' threw exception; nested exception is org.redisson.client.RedisConnectionException: Unable to connect to Redis server: localhost/127.0.0.1:6379



Thanks
